Tests/NIOWebSocketTests/NIOWebSocketFrameAggregatorTests.swift

```diff
@@ -0,0 +1,156 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import XCTest
+import NIO
+import NIOWebSocket
+
+final class NIOWebSocketFrameAggregatorTests: XCTestCase {
+    var channel: EmbeddedChannel!
+    
+    override func setUp() {
+        channel = EmbeddedChannel(
+            handler: NIOWebSocketFrameAggregator(
+                minNonFinalFragmentSize: 1,
+                maxAccumulatedFrameCount: 4,
+                maxAccumulatedFrameSize: 32
+            )
+        )
+    }
+    
+    override func tearDown() {
+        XCTAssertEqual(try self.channel.finish().isClean, true)
+    }
+    
+    func testEmptyButFinalFrameIsForwardedEvenIfMinNonFinalFragmentSizeIsGreaterThanZero() throws {
+        let frame = WebSocketFrame(fin: true, opcode: .binary, data: ByteBuffer())
+        try channel.writeInbound(frame)
+        XCTAssertEqual(try channel.readInbound(as: WebSocketFrame.self), frame)
```

Comment:
```suggestion
    func testEmptyButFinalFrameIsForwardedEvenIfMinNonFinalFragmentSizeIsGreaterThanZero() {
        let frame = WebSocketFrame(fin: true, opcode: .binary, data: ByteBuffer())
        XCTAssertNoThrow(try channel.writeInbound(frame))
        XCTAssertEqual(try channel.readInbound(as: WebSocketFrame.self), frame)
    }
```

One nit that I've just seen. All added tests potentially throw. Those throwing tests sadly have unclear error messages in case they actually throw, which is unfortunate in CI setups. Would you mind wrapping all `try`s in an `XCTAssertNoThrow`?

Reply:
Done. Hower, Xcode shows me the line which throws e.g.:
<img width="510" alt="Screenshot 2021-04-29 at 11 31 59" src="https://user-images.githubusercontent.com/5688874/116530706-9f139980-a8de-11eb-90ca-aae394064e5f.png">
Is this just a limitation of CLI or Linux?

Reply:
I believe the answer is "both".